### PR TITLE
Add Apple Silicon support

### PR DESCRIPTION
**Pull request by lunij**
This is an addition to the PR #12 by @nemoDreamer.

> Newer installations of Homebrew use /opt/homebrew/ instead of /usr/local/.

The difference is the `if type -q brew` condition that didn't work for me on my M1. `type` can only find `brew` when it is already part of the `$PATH` variable, but to be part of the `$PATH` variable this fish script needs to run first.

In other words: This fish script adds necessary paths used by `brew` to `$PATH` to make `brew` accessible. Only after this script did its job, commands like `type brew` and `which brew` can finally work.

### CODE REVIEW

**Review by scorphus — APPROVED** [review]
Hey, @lunij, thanks for contributing!

How about reporting the absence of `brew` at the end, after all existing paths are appended to `$PATH`? This may not only maintain the behavior in that regard but also help troubleshooting issues.

**Comment by lunij** [comment]
> How about reporting the absence of brew at the end [...]

Yes, I did it.

In addition I changed the algorithm to one `for loop` only. Now, a path is added to `$PATH` only if it is not already part of `$PATH` and contains a file called `brew`. Also, the path is added in the beginning of `$PATH` to ensure it has precedence over system paths, otherwise system apps like `ruby` and `python` would be preferred over the Homebrew variants. It should be the other way around.

**Comment by scorphus** [comment]
> > How about reporting the absence of brew at the end [...]
> 
> Yes, I did it.

Thank you!

> In addition I changed the algorithm to one `for loop` only. Now, a path is added to `$PATH` only if it is not already part of `$PATH` and contains a file called `brew`. [...]

Nice! We need to make sure `contains` is supported in older Fish releases.

> [...] Also, the path is added in the beginning of `$PATH` to ensure it has precedence over system paths, otherwise system apps like `ruby` and `python` would be preferred over the Homebrew variants. It should be the other way around.

It's usually a good idea to give system apps preference. Perhaps not when it's the user's shell (whereas in bash and scripts not child of the user's shell process, system's apps would still be preferred — which could also cause confusion other than failures).

I don't really know all implications as I always append (instead of prepend) brew path to PATH — and use virtual environments for pretty much everything (ruby, python, etc.)

Still, this is a radical change that might affect many users who have it the other way around. Maybe that should still be the default behavior and a the plugin could support some sort of a flag to indicate that brew paths should be prepended.

What are your thoughts, @bobthecow?

File: conf.d/brew.fish [hunk 1]
```diff
@@ -1,24 +1,17 @@
...
-    end
+  if test -f $brew_path/brew
+    set brew_found true
+    set PATH $brew_path $PATH
+    break
   end
-else
+end
+
+if not set -q brew_found
```
Comment (antisaling) [inline conf.d/brew.fish]: My suggestion would be:

```suggestion
if not set -q brew_found; and not command -q brew
```

**Review by antisaling — COMMENTED** [review]
Hi, folks. I've been trying to set this up on my machine and was having issues until I noticed that if for some reason, the user is already appending to `PATH` somewhere else, this script will yell needlessly.

**Comment by misimpso** [comment]
I was having the problem of brew not being available on the path, and fish complaining with the "type -q" conditional check failing. This fix, along with the first fix from this [issue](https://github.com/oh-my-fish/plugin-brew/issues/14#issuecomment-998102135), fixed my problem.

**Review by scorphus — APPROVED** [review]

**Comment by scorphus** [comment]
Thanks, @lunij! :shipit:

**Pull request merged**